packages/contractkit/src/wrappers/BaseWrapper.ts

```diff
@@ -50,19 +50,22 @@ export const valueToInt = (input: BigNumber.Value) =>
 export const valueToFrac = (numerator: BigNumber.Value, denominator: BigNumber.Value) =>
   valueToBigNumber(numerator).div(valueToBigNumber(denominator))
 
-export const stringToBuffer = hexToBuffer
-
-export const bufferToString = (buf: Buffer) => ensureLeading0x(buf.toString('hex'))
-
-type SolBytes = string | number[]
-const toBytes = (input: any): SolBytes => input
-const fromBytes = (input: SolBytes): any => input as any
-
-export const stringToBytes = (input: string) => toBytes(ensureLeading0x(input))
-
-export const bufferToBytes = (input: Buffer) => stringToBytes(bufferToString(input))
-
-export const bytesToString = (input: SolBytes): string => fromBytes(input)
+// Type of bytes in solidity gets repesented as a string of number array by typechain and web3
+// Hopefull this will improve in the future, at which point we can make improvements here
+type SolidityBytes = string | number[]
+export const stringToSolidityBytes = (input: string) => ensureLeading0x(input) as SolidityBytes
+export const bufferToSolidityBytes = (input: Buffer) => stringToSolidityBytes(bufferToHex(input))
+export const solidityBytesToString = (input: SolidityBytes): string => {
+  if (input === null || input === undefined || typeof input === 'string') {
+    return input
+  } else if (Array.isArray(input)) {
+    let hex = ''
+    input.map((num) => (hex += num))
```

Comment:
(Detail)
Although this work, you are using a `map` as an iterator, and the essence is different.
You can use `forEach` instead of `map` for the same result, or even `reduce`
```
const hex = input.reduce((accum, num) => accum + num, '')
```
With reduce it's clearer that you are transforming an array into a result

Reply:
I'm not a big fan of `reduce` in JS, i find the interface unintuitive, but agreed in this case it would work nicely. I'll make that change when we decide what to do with this section